BuilderDesignPattern/Program.cs

```diff
@@ -0,0 +1,411 @@
+
+        //    public Director(IPizzaBuilder pizzaBuilder)
+        //    {
+        //        _pizzaBuilder = pizzaBuilder;
+        //    }
+
+        //    public void MakeCustomPizza(string dough, string sauce, List<string> topping)
+        //    {
+        //        if (!string.IsNullOrEmpty(dough)) _pizzaBuilder.BuildDough(dough);
+        //        if (!string.IsNullOrEmpty(sauce)) _pizzaBuilder.BuildSauce(sauce);
+        //        if (topping != null && topping.Count > 0) _pizzaBuilder.BuildTopping(topping);
+        //    }
+        //}
+        #endregion
+
+        static void Main(string[] args)
+        {
+            #region Task #1
+            IPizzaBuilder pizzaBuilder = new WhitePizzeBuilder();
+            Director director = new Director(pizzaBuilder);
+
+            // white pizza
+            director.MakePizza();
+            Pizza pizza = (pizzaBuilder as WhitePizzeBuilder).GetPizza();
+            Console.WriteLine(pizza);
+
+            // white pizza with topping
+            director.MakePizzaWithTopping();
+            pizza = (pizzaBuilder as WhitePizzeBuilder).GetPizza();
+            Console.WriteLine(pizza);
+
+            // wheat pizza
+            director.PizzaBuilder = (pizzaBuilder = new WheatPizzaBuilder());
+            director.MakePizza();
+            pizza = (pizzaBuilder as WheatPizzaBuilder).GetPizza();
+            Console.WriteLine(pizza);
+
+            // wheat pizza with topping
+            director.MakePizzaWithTopping();
+            pizza = (pizzaBuilder as WheatPizzaBuilder).GetPizza();
+            Console.WriteLine(pizza);
+            #endregion
+
+
+            #region Bouns 1
+            //IPizzaBuilder pizzaBuilder = new WhitePizzeBuilder();
+
+            //// white pizza
+            //pizzaBuilder.BuildDough();
+            //pizzaBuilder.BuildSauce();
+            //Pizza pizza = (pizzaBuilder as WhitePizzeBuilder).GetPizza();
+            //Console.WriteLine(pizza);
+
+            //// white pizza with topping
+            //pizzaBuilder.BuildDough();
+            //pizzaBuilder.BuildSauce();
+            //pizzaBuilder.BuildTopping();
+            //pizza = (pizzaBuilder as WhitePizzeBuilder).GetPizza();
+            //Console.WriteLine(pizza);
+
+            //// wheat pizza
+            //pizzaBuilder = new WheatPizzaBuilder();
+            //pizzaBuilder.BuildDough();
+            //pizzaBuilder.BuildSauce();
+            //pizza = (pizzaBuilder as WheatPizzaBuilder).GetPizza();
+            //Console.WriteLine(pizza);
+
+            //// wheat pizza with topping
+            //pizzaBuilder.BuildDough();
+            //pizzaBuilder.BuildSauce();
+            //pizzaBuilder.BuildTopping();
+            //pizza = (pizzaBuilder as WheatPizzaBuilder).GetPizza();
+            //Console.WriteLine(pizza);
+            #endregion
+
+
+            #region Bouns 2 | Comment out (#region Task #1 & Bouns #1) and uncomment (#region Bouns #2)
```

Comment:
I would like to thank you for doing the second Bonus with dedication! You did it with a great effort👏